include/unifex/when_all.hpp

```diff
@@ -33,221 +33,264 @@
+    ops_.start();
+  }
 
-   private:
-    void element_complete() noexcept {
-      if (refCount_.fetch_sub(1, std::memory_order_acq_rel) == 1) {
-        deliver_result();
-      }
+  private:
+  void element_complete() noexcept {
+    if (refCount_.fetch_sub(1, std::memory_order_acq_rel) == 1) {
+      deliver_result();
     }
+  }
 
-    void deliver_result() noexcept {
-      stopCallback_.destruct();
-
-      if (get_stop_token(receiver_).stop_requested()) {
-        unifex::set_done(std::move(receiver_));
-      } else if (doneOrError_.load(std::memory_order_relaxed)) {
-        if (error_.has_value()) {
-          std::visit(
-              [this](auto&& error) {
-                unifex::set_error(std::move(receiver_), (decltype(error))error);
-              },
-              std::move(error_.value()));
-        } else {
-          unifex::set_done(std::move(receiver_));
-        }
+  void deliver_result() noexcept {
+    stopCallback_.destruct();
+
+    if (get_stop_token(receiver_).stop_requested()) {
+      unifex::set_done(std::move(receiver_));
+    } else if (doneOrError_.load(std::memory_order_relaxed)) {
+      if (error_.has_value()) {
+        std::visit(
+            [this](auto&& error) {
+              unifex::set_error(std::move(receiver_), (decltype(error))error);
+            },
+            std::move(error_.value()));
       } else {
-        deliver_value(std::index_sequence_for<Senders...>{});
+        unifex::set_done(std::move(receiver_));
       }
+    } else {
+      deliver_value(std::index_sequence_for<Senders...>{});
     }
+  }
 
-    template <std::size_t... Indices>
-    void deliver_value(std::index_sequence<Indices...>) noexcept {
-      try {
-        unifex::set_value(
-            std::move(receiver_),
-            std::get<Indices>(std::move(values_)).value()...);
-      } catch (...) {
-        unifex::set_error(std::move(receiver_), std::current_exception());
-      }
+  template <std::size_t... Indices>
+  void deliver_value(std::index_sequence<Indices...>) noexcept {
+    try {
+      unifex::set_value(
+          std::move(receiver_),
+          std::get<Indices>(std::move(values_)).value()...);
+    } catch (...) {
+      unifex::set_error(std::move(receiver_), std::current_exception());
     }
+  }
 
-    std::tuple<std::optional<
-        typename Senders::template value_types<std::variant, std::tuple>>...>
-        values_;
-    std::optional<error_types<std::variant>> error_;
-    std::atomic<std::size_t> refCount_{sizeof...(Senders)};
-    std::atomic<bool> doneOrError_{false};
-    inplace_stop_source stopSource_;
-    UNIFEX_NO_UNIQUE_ADDRESS manual_lifetime<typename stop_token_type_t<
-        Receiver&>::template callback_type<cancel_operation>>
-        stopCallback_;
-    Receiver receiver_;
-    detail::when_all_operation_tuple<0, element_receiver, Senders...> ops_;
-  };
+  std::tuple<std::optional<
+      typename Senders::template value_types<std::variant, std::tuple>>...>
```

Comment:
Hmm, looking at this again I wonder if it should be storing a tuple of decay-copied values instead of just the value-types directly?

Reply:
I'll leave it for now, and let you make the change later.